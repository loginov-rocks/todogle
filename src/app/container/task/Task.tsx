import * as React from 'react';

const Task = () => (
  <strong>Task</strong>
);

export default Task;
