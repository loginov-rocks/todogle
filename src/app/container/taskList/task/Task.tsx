import { Button } from '@material-ui/core';
import * as React from 'react';
import { useHistory } from 'react-router-dom';

import * as R from '../../../../routes';
import gapi from '../../../../services/gapi';
import { TaskListResource } from '../../../../services/gapi/TaskListResource';
import { TaskResource } from '../../../../services/gapi/TaskResource';

interface Props {
  onDelete: (id: string) => void;
  task: TaskResource;
  taskList: TaskListResource;
}

const Task = ({ onDelete, task, taskList }: Props) => {
  const history = useHistory();

  const handleClick = () => {
    history.push(R.toTask(taskList.id, task.id));
  };

  const handleDelete = () => {
    gapi.deleteTask(taskList.id, task.id)
      .then(() => onDelete(task.id));
  };

  return (
    <div>
      <i onClick={handleClick}>{task.title}</i>
      <Button onClick={handleDelete} variant="contained">Delete</Button>
    </div>
  );
};

export default Task;
