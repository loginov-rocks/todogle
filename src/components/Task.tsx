import * as React from 'react';

import gapi from '../services/gapi';
import { TaskListResource } from '../services/gapi/TaskListResource';
import { TaskResource } from '../services/gapi/TaskResource';

interface Props {
  onDelete: (id: string) => void;
  task: TaskResource;
  taskList: TaskListResource;
}

export default class Task extends React.Component<Props> {
  constructor(props: Props) {
    super(props);

    this.handleDelete = this.handleDelete.bind(this);
  }

  handleDelete() {
    const { onDelete, task, taskList } = this.props;

    gapi.deleteTask(taskList.id, task.id)
      .then(() => onDelete(task.id));
  }

  render() {
    const { task } = this.props;

    return (
      <div>
        <i>{task.title}</i>
        <button onClick={this.handleDelete}>Delete</button>
      </div>
    );
  }
}
