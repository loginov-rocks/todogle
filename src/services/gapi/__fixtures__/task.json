{
  "kind": "tasks#task",
  "id": "SUxBbWpKTkZNd1J3ZGJkSw",
  "etag": "\"MTA0MzI0OTMyNg\"",
  "title": "The best task",
  "updated": "2020-07-11T14:01:56.000Z",
  "selfLink": "https://www.googleapis.com/tasks/v1/lists/RUVUM3JvcDUxTUNHcDUtdw/tasks/SUxBbWpKTkZNd1J3ZGJkSw",
  "position": "00000000000000000000",
  "status": "needsAction"
}
