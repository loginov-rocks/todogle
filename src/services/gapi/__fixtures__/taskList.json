{
  "kind": "tasks#taskList",
  "id": "RUVUM3JvcDUxTUNHcDUtdw",
  "etag": "\"MTA0MzI0OTM1Nw\"",
  "title": "The best task list",
  "updated": "2020-07-11T14:01:56.445Z",
  "selfLink": "https://www.googleapis.com/tasks/v1/users/@me/lists/RUVUM3JvcDUxTUNHcDUtdw"
}
