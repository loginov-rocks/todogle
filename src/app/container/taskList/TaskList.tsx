import { Button, CircularProgress } from '@material-ui/core';
import * as React from 'react';

import gapi from '../../../services/gapi';
import { TaskListResource } from '../../../services/gapi/TaskListResource';
import { TaskResource } from '../../../services/gapi/TaskResource';

import CreateTask from './createTask/CreateTask';
import Task from './task/Task';

interface Props {
  onDelete: (id: string) => void;
  taskList: TaskListResource;
}

const TaskList = ({ onDelete, taskList }: Props) => {
  const [areTasksLoaded, setAreTasksLoaded] = React.useState(false);
  const [tasks, setTasks] = React.useState<TaskResource[]>([]);

  React.useEffect(() => {
    setAreTasksLoaded(false);

    gapi.getTasks(taskList.id)
      .then(tasks => {
        setAreTasksLoaded(true);
        setTasks(tasks);
      });
  }, [taskList.id]);

  const handleDelete = () => {
    gapi.deleteTaskList(taskList.id)
      .then(() => onDelete(taskList.id));
  };

  const handleTaskCreate = (task: TaskResource) => {
    setTasks(tasks.concat([task]));
  };

  const handleTaskDelete = (taskId: string) => {
    setTasks(tasks.filter(task => task.id !== taskId));
  };

  return (
    <>
      <b>{taskList.title}</b>
      <Button onClick={handleDelete} variant="contained">Delete</Button>

      {areTasksLoaded ? (
        <>
          <CreateTask
            onCreate={handleTaskCreate}
            taskListId={taskList.id}
          />

          {tasks.map(task => (
            <Task
              key={task.id}
              onDelete={handleTaskDelete}
              task={task}
              taskList={taskList}
            />
          ))}
        </>
      ) : (
        <CircularProgress />
      )}
    </>
  );
};

export default TaskList;
