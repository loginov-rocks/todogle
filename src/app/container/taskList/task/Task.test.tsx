import { shallow } from 'enzyme';
import * as React from 'react';

import task from '../../../../services/gapi/__fixtures__/task.json';
import taskList from '../../../../services/gapi/__fixtures__/taskList.json';

import Task from './Task';

it('matches snapshot', () => {
  const wrapper = shallow(
    <Task onDelete={() => undefined} task={task} taskList={taskList} />,
  );

  expect(wrapper).toMatchSnapshot();
});
