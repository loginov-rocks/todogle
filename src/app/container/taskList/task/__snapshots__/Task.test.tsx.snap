// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <WithStyles(ForwardRef(Checkbox))
    checked={false}
    onClick={[Function]}
  />
  <i
    onClick={[Function]}
  >
    The best task
  </i>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
    variant="contained"
  >
    Delete
  </WithStyles(ForwardRef(Button))>
</div>
`;
